### PR TITLE
cherry-pick from upstream jemalloc to fix profiling bug

### PR DESCRIPTION
**Pull request by umanwizard**
This depends on https://github.com/tikv/jemalloc/pull/2 , which is a cherry-pick of https://github.com/jemalloc/jemalloc/pull/1604 .

See https://github.com/jemalloc/jemalloc/issues/1605 for details of the bug that this fixes; basically, it makes profiling/sampling fail to work under some conditions. We hit this in practice at Materialize when attempting to switch to tikv/jemallocator.

### CODE REVIEW

**Comment by umanwizard** [comment]
@BusyJay , @sticnarf , this is ready for a look now that tikv/jemalloc#2 has landed.

**Comment by BusyJay** [comment]
Travis CI is out of quota, I may need some time to migrate to github actions.

File: jemalloc-sys/configure/configure [hunk 1]
```diff
@@ -862,6 +863,7 @@ datadir='${datarootdir}'
 sysconfdir='${prefix}/etc'
 sharedstatedir='${prefix}/com'
 localstatedir='${prefix}/var'
+runstatedir='${localstatedir}/run'
```
Comment (BusyJay) [inline jemalloc-sys/configure/configure:866]: Why the change?

Comment (umanwizard) [reply jemalloc-sys/configure/configure:866]: I updated the configure file (`cd jemalloc-sys/jemalloc && autoconf && cp configure ../configure/configure`) as a matter of habit when modifying autoconf-based sub modules. We can remove it if you prefer.

Comment (BusyJay) [reply jemalloc-sys/configure/configure:866]: I think it's related to the version of autoconf. Unless it can fail compilation, better leave them untouch.

**Comment by BusyJay** [comment]
New CI can be triggered once it's merged master.

**Review by BusyJay — APPROVED** [review]
Thanks!

**Comment by umanwizard** [comment]
Unfortunately, this broke tikv/jemallocator, since tikv/jemalloc#2 landed with a different commit number. Sending a new PR now.

**Comment by umanwizard** [comment]
Fixed in https://github.com/tikv/jemallocator/pull/12
